### PR TITLE
Fix compatibility with `league/flysystem-bundle` `3.3.0`

### PR DESCRIPTION
**Pull request by fritzmg**
Fixes https://github.com/contao/conflicts/pull/50

### CODE REVIEW

File: core-bundle/src/DependencyInjection/Filesystem/FilesystemConfiguration.php [hunk 1]
```diff
@@ -88,7 +88,7 @@ public function mountAdapter(string $adapter, array $options, string $mountPath,
         $name ??= str_replace(['.', '/', '-'], '_', Container::underscore($mountPath));
         $adapterId = "contao.filesystem.adapter.$name";
 
-        if (null !== ($adapterDefinition = $this->adapterDefinitionFactory->createDefinition($adapter, $options))) {
+        if (null !== ($adapterDefinition = $this->adapterDefinitionFactory->createDefinition($adapter, $options, null))) {
```
Comment (ausi) [inline core-bundle/src/DependencyInjection/Filesystem/FilesystemConfiguration.php:91]: Other possibilities would be
```suggestion
        if (null !== ($adapterDefinition = $this->adapterDefinitionFactory->createDefinition($adapter, $options, Visibility::PUBLIC))) {
```
or:
```suggestion
        if (null !== ($adapterDefinition = $this->adapterDefinitionFactory->createDefinition($adapter, $options, Visibility::PRIVATE))) {
```

Togehter with a `use League\Flysystem\Visibility;`

Or adding a parameter to the method that gets passed on?

Maybe @m-vo can help here?

Comment (fritzmg) [reply core-bundle/src/DependencyInjection/Filesystem/FilesystemConfiguration.php:91]: Passing `null` seems to restore the previous defaults, as far as I can see. However this definitely needs a review.

**Comment by ausi** [comment]
Once this is merged and released we should not forget to revert https://github.com/contao/conflicts/pull/51

File: core-bundle/src/DependencyInjection/Filesystem/FilesystemConfiguration.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/DependencyInjection/Filesystem/FilesystemConfiguration.php:91]: It seems that the new argument will be optional in the next release: https://github.com/thephpleague/flysystem-bundle/pull/156

Comment (fritzmg) [reply core-bundle/src/DependencyInjection/Filesystem/FilesystemConfiguration.php:91]: Ah nice. I guess we can close this PR and https://github.com/contao/conflicts/pull/51

Comment (ausi) [reply core-bundle/src/DependencyInjection/Filesystem/FilesystemConfiguration.php:91]: So we don’t need to do anything, right? 🎉

Comment (m-vo) [reply core-bundle/src/DependencyInjection/Filesystem/FilesystemConfiguration.php:91]: For the record: if things like this turn out to be a problem in the future, we could also remove the dependency to `league/flysystem-bundle` completely and handle things ourselves. Right now we are more or less just using the the `AdapterDefinitionFactory` which helps to turn the (rather specific) adapter config into services.

**Pull request closed without merging**